### PR TITLE
Fv upload

### PR DESCRIPTION
**Pull request by andrewbutterfield**
Here are the changes to add in the Formal Verification material to rtems-central.

The archive material has been left out.

There are two portions of 3rd-party code: `comment_filter` and `promela_yacc`, both in `formal/promela/src/src/modules`.
Each has two commits - the original material imported in, and how we changed it.

I'd like to make changes that remove the need for `comment_filter`.  I also think that the promela stuff can now be got via `pip`.

### CODE REVIEW

File: formal/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Formal Verification
+
+`formal`
+
+This directory contains the models and tooling developed as part of the ESA-sponsored activity ***Qualification of RTEMS Symmetric Multiprocessing(SMP)***, that has been added into RTEMS in the `rtems-central` repository.
```
Comment (sebhub) [inline formal/README.md:5]: There is a strange character in this line.

File: formal/promela/models/threadq/docs/notes/Intro.tex [hunk 2]
```diff
@@ -0,0 +1,7 @@
+\section{Introduction}
```
Comment (sebhub) [inline formal/promela/models/threadq/docs/notes/Intro.tex:1]: The integration of this copy and paste stuff from various sources is a bit problematic. Also the files in the "doc" directory have not SPDX Licence Identifier. I think the integration of the documentation should be done at a later stage and maybe to the rtems-docs repository.

File: formal/promela/models/threadq/srccpy/mrsp-scenarios/mrsp-Init.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/*
```
Comment (sebhub) [inline formal/promela/models/threadq/srccpy/mrsp-scenarios/mrsp-Init.c:1]: Do we really need the "srccpy" directory?

File: formal/README.md (hunk 1 above)
Comment (andrewbutterfield) [reply formal/README.md:5]: Ouch! I'll fix that.

File: formal/promela/models/threadq/docs/notes/Intro.tex (hunk 2 above)
Comment (andrewbutterfield) [reply formal/promela/models/threadq/docs/notes/Intro.tex:1]: I agree - most of this is for my own use, anyway.

File: formal/promela/models/threadq/srccpy/mrsp-scenarios/mrsp-Init.c (hunk 3 above)
Comment (andrewbutterfield) [reply formal/promela/models/threadq/srccpy/mrsp-scenarios/mrsp-Init.c:1]: No, we don't.

**Comment by sebhub** [comment]
All commits except "Thread Queue models (work in progress)" were integrated.

**Pull request closed without merging**